src/contrib/cluster/Akka.Cluster.Tools.Tests.MultiNode/PublishSubscribe/DistributedPubSubMediatorSpec.cs

```diff
@@ -718,17 +719,18 @@ public void DistributedPubSubMediator_must_transfer_delta_correctly()
                     Mediator.Tell(new Put(CreateChatUser("u" + (1000 + i))));
                 }
 
-                Mediator.Tell(new Tools.PublishSubscribe.Internal.Status(new Dictionary<Address, long>(), isReplyToStatus: false));
+                Mediator.Tell(new Tools.PublishSubscribe.Internal.Status(ImmutableDictionary<Address, long>.Empty, isReplyToStatus: false));
                 var deltaBuckets1 = ExpectMsg<Delta>().Buckets;
-                deltaBuckets1.Sum(x => x.Content.Count).ShouldBe(500);
+                deltaBuckets1.Sum(x => x.Content.Count).Should().Be(500);
+                var versions1 = deltaBuckets1.ToImmutableDictionary(b => b.Owner, b => b.Version);
 
-                Mediator.Tell(new Tools.PublishSubscribe.Internal.Status(deltaBuckets1.ToDictionary(b => b.Owner, b => b.Version), isReplyToStatus: false));
+                Mediator.Tell(new Tools.PublishSubscribe.Internal.Status(versions1, isReplyToStatus: false));
                 var deltaBuckets2 = ExpectMsg<Delta>().Buckets;
-                deltaBuckets2.Sum(x => x.Content.Count).ShouldBe(500);
+                deltaBuckets2.Sum(x => x.Content.Count).Should().Be(500);
 
-                Mediator.Tell(new Tools.PublishSubscribe.Internal.Status(deltaBuckets2.ToDictionary(b => b.Owner, b => b.Version), isReplyToStatus: false));
+                Mediator.Tell(new Tools.PublishSubscribe.Internal.Status(versions1.SetItems(deltaBuckets2.ToImmutableDictionary(b => b.Owner, b => b.Version)), isReplyToStatus: false));
```

Comment:
this test had random behavior, depending on the order in dictionary. In some cases it lost the version info from the previous runs and returned data multiple times

Reply:
Nice fix - this one has definitely been racy in the past.